File: grida-domains/domain-image/src/main/java/org/grida/diaryimage/DiaryAnalyzer.java
package org.grida.diaryimage;

import lombok.RequiredArgsConstructor;
import org.grida.processor.diaryanalyze.DiaryAnalyzeProcessor;
import org.grida.processor.diaryanalyze.DiaryAnalyzeResult;
import org.grida.prompt.PromptKeywords;
import org.grida.prompt.PromptProvider;
import org.springframework.stereotype.Component;

import static org.grida.prompt.Prompt.DIARY_ANALYSIS;

@Component
@RequiredArgsConstructor
public class DiaryAnalyzer {

    private final PromptProvider promptProvider;
    private final DiaryAnalyzeProcessor diaryAnalyzeProcessor;

    public DiaryAnalyzeResult analyze(PromptKeywords keywords) {
        String prompt = promptProvider.provide(DIARY_ANALYSIS, keywords);
        return diaryAnalyzeProcessor.proceed(prompt);
    }
}

File: grida-core/core-llm/src/main/kotlin/org/grida/provider/diaryimage/DiaryAnalyzer.kt
package org.grida.provider.diaryimage

import org.grida.model.chat.ChatModel
import org.springframework.stereotype.Component

@Component
class DiaryAnalyzer(
    private val chatModel: ChatModel,
    private val promptBuilder: DiaryAnalyzePromptBuilder
) {

    fun analyze(key: DiaryImageKey): AnalyzedDiary {
        val prompt = promptBuilder.build(key)
        val result = chatModel.call(prompt, AnalyzedDiary::class.java)
        return result
    }
}
